engine/table/src/main/java/io/deephaven/engine/table/impl/locations/TableDataService.java

```diff
@@ -20,6 +20,24 @@ public interface TableDataService {
     @NotNull
     TableLocationProvider getTableLocationProvider(@NotNull TableKey tableKey);
 
+    /**
+     * Request the raw {@link TableLocationProvider} from this service that will provide the {@link TableLocation} for
+     * {@code tableKey} and {@code tableLocationKey} if the location may exist. A raw {@link TableLocationProvider} does
```

Comment:
I believe this is still too vague. If you are dealing with "may exist", then you have to handle multiples. I think this needs to be defined and implemented as returning the single RTLP that _does_ currently hold the requested location (and a TableDataException if it is not singular).